File: Backend/Online-Store-Backend/Online-Store-Backend.Database/Employees/Models/Employee.cs
using Online_Store_Backend.Database.Entity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace Online_Store_Backend.Database.Employees.Models
{
    [Serializable]
    [DataContract]
    [Table("employee", Schema = "public")]
    public class Employee : BaseEntity
    {
        [DataMember]
        [MaxLength(50)]
        [Required]
        public string FirstName { get; set; }

        [DataMember]
        [MaxLength(50)]
        [Required]
        public string? LastName { get; set; }

        [DataMember]
        [MaxLength(20)]
        [Required]
        public string? PhoneNum { get; set; }

        [DataMember]
        [MaxLength(100)]
        [Required]
        public string? Email { get; set; }

        [DataMember]
        public bool? Gender { get; set; }

        [DataMember]
        public DateTime? BirthDate { get; set; }

        [DataMember]
        [MaxLength(255)]
        public string Login { get; set; }

        [DataMember]
        [MaxLength(255)]
        public string? PasswordHash { get; set; }
    }
}

File: Frontend/Online-Store-App.CommonLib/src/Models/Data/Employee.ts
import { type BaseEntity } from "./Base/BaseEntity";

export interface Employee extends BaseEntity {
    firstName: string;
    lastName: string;
    phoneNum: string;
    email: string;
    gender: boolean | undefined;
    birthDate: Date;
    login: string;
}

export interface EmployeeForPost {
    firstName: string;
    lastName: string | undefined;
    phoneNum: string | undefined;
    email: string | undefined;
    gender: boolean | undefined;
    birthDate: Date | undefined;
    login: string;
    password: string;
    confirmPassword: string;
} 

export interface EmployeeForPut {
    id: number;
    firstName: string;
    lastName: string;
    phoneNum: string;
    email: string;
    gender: boolean | undefined;
    birthDate: Date;
    login: string;
} 